gapic/schema/api.py

```diff
@@ -526,24 +532,73 @@ def _get_methods(self,
                         'metadata type.',
                     )
                 lro = wrappers.OperationInfo(
-                    response_type=self.api_messages[address.resolve(
+                    response_type=self.api_messages[service_address.resolve(
                         op.response_type,
                     )],
-                    metadata_type=self.api_messages[address.resolve(
+                    metadata_type=self.api_messages[service_address.resolve(
                         op.metadata_type,
                     )],
                 )
 
+            # If we got a gRPC service config, get the appropriate retry
+            # and timeout information from it.
+            retry = None
+            timeout = None
+            if self.opts.retry:
+                # This object should be a dictionary that conforms to the
+                # gRPC service config proto:
+                #   Repo: https://github.com/grpc/grpc-proto/
+                #   Filename: grpc/service_config/service_config.proto
+                #
+                # We only care about a small piece, so we are just leaving
+                # it as a dictionary and parsing accordingly.
+                for mc in self.opts.retry.get('methodConfig', []):
+                    # Does this method config apply to us?
+                    # The gRPC service config uses a repeated `name` field
+                    # with a particular format, which we match against.
+                    selector = {
+                        'service': '{package}.{service_name}'.format(
+                            package='.'.join(service_address.package),
+                            service_name=service_address.name,
+                        ),
+                        'method': meth_pb.name,
+                    }
```

Comment:
This structure should be lifted up; if you stick with the current control flow, it's a loop invariant, and if you change it, it's a data dependency for the `next` expression.

Reply:
Done, and added a better comment.